app/services/foreman/plugin.rb

```diff
@@ -83,6 +83,12 @@ def find(id)
         registered_plugins[id.to_sym]
       end
 
+      # Evaluates additional setup steps after initial registration
+      # useful when you need to define methods later e.g. from config.to_prepare
+      def setup(id, &block)
```

Comment:
Would it make sense to call it from other places than from `to_prepare`. Why? I understand we're adding it for this particular case. Another thing is it might lead to some unexpected behaviour. when doing too late (as in this block one could use all the plugin dsl, even the one that would be to late to change). Therefore I would suggest to make this extension point really limited just extendnig template helpers.

Another approach would be to store the template helpers as string, which would mean we could perform the constantization later in foreman, without the plugin developer would even need to care and user the `setup` at all